lib/gb-tracker-core.js

```diff
@@ -380,7 +380,7 @@ const TrackerFactory = function(__schemas__, __sanitizeEvent__) {
       const SEGMENT_WRAPPER_OVERHEAD = encodeURIComponent(JSON.stringify(segmentTemplate)).length;
 
       // Double encode here to account for double-encoding at the end
-      const eventStringSegments = utils.chunkString(eventString, MAX_QUERY_STRING_LENGTH - SEGMENT_WRAPPER_OVERHEAD);
+      const eventStringSegments = utils.chunkString(eventString, MAX_QUERY_STRING_LENGTH - SEGMENT_WRAPPER_OVERHEAD, LZString.compressToEncodedURIComponent);
 
       if (eventStringSegments.length > MAX_SEGMENT_COUNT) {
         console.error(`cannot send: ${ eventStringSegments } event segments, as that exceeds the max of: ${ MAX_SEGMENT_COUNT}`);
```

Comment:
There is still a call to  LZString.compressToEncodedURIComponent below on each segment

Reply:
The `chunkString` function does not encode the chunks - it only uses the encoder to check if the length is within the bounds or not. After you got the eventStringSegments, you still have to encode them